File: src/main/python/com/hazza/LeetCodeByPython/SegmentOne/No27_Remove_Element.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019/4/22 20:52
# https://leetcode.com/problems/remove-element/
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if not nums or len(nums) == 0:
            return 0

        cnt = 0
        for i in range(0, len(nums)):
            if nums[i] != val:
                nums[cnt] = nums[i]
                cnt += 1

        return cnt


if __name__ == '__main__':
    pass

File: src/main/java/com/hazza/leetcode/SegmentOne/No27RemoveElement.java
package com.hazza.leetcode.SegmentOne;

/**
 * Created by hazza on 7/14/17.
 */
public class No27RemoveElement {
    public int removeElement(int[] nums, int val) {
        if (nums == null || nums.length == 0) return 0;
        int count = 0, len = nums.length;
        for (int i = 0; i < len; i++)
            if (nums[i] != val) nums[count++] = nums[i];

        return count;
    }
}
